File: gm-utils/src/main/java/src/commom/utils/doubles/DoubleParse.java
package src.commom.utils.doubles;

import js.Js;

public class DoubleParse {

	public static Double toDouble(Object o) {
		if (DoubleIs.is(o)) {
			return Js.parseFloat(o);
		}
		throw new Error("NaN");
	}

	public static Double toDoubleDef(Object o, double def) {
		if (DoubleIs.is(o)) {
			return Js.parseFloat(o);
		}
		return def;
	}

}

File: gm-utils/js/src/commom/utils/doubles/DoubleParse.js
import DoubleIs from './DoubleIs';

export default class DoubleParse {

	static toDouble(o) {
		if (DoubleIs.is(o)) {
			return parseFloat(o);
		}
		throw new Error("NaN");
	}

	static toDoubleDef(o, def) {
		if (DoubleIs.is(o)) {
			return parseFloat(o);
		}
		return def;
	}

}
